backend/src/main/java/com/google/sps/servlets/AddParticipantServlet.java

```diff
@@ -99,23 +87,35 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
       return;
     }
 
-    // id is irrelevant, only relevant when getting from datastore
+    // Create new Participant from input parameters
+    // id is irrelevant, created when added to datastore
     Participant newParticipant =
         new Participant(
-            /* id= */ -1L, username, startTimeAvailable, endTimeAvailable, duration, timestamp);
+            username, startTimeAvailable, endTimeAvailable, duration, /* matchId=*/ 0, timestamp);
 
+    // Get DatastoreService and instiate Match and Participant Datastores
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    MatchDatastore matchDatastore = new MatchDatastore(datastore);
+    ParticipantDatastore participantDatastore = new ParticipantDatastore(datastore);
 
     // Find immediate match if possible
-    FindMatchQuery query = new FindMatchQuery(Clock.systemUTC());
-    Match match = query.findMatch(getParticipants(datastore), newParticipant);
+    FindMatchQuery query = new FindMatchQuery(Clock.systemUTC(), participantDatastore);
+    Match match = query.findMatch(newParticipant);
 
-    // Match found, add to datastore, delete matched participants from datastore
     if (match != null) {
-      addMatchToDatastore(match, datastore);
-      deleteParticipantFromDatastore(match.getSecondParticipant(), datastore);
+      // Match found, add to match datastore, update participant datastore
+      long matchId = matchDatastore.addMatch(match);
+
+      System.out.println(matchDatastore.getMatchFromId(matchId).toString());
+
+      // Update participant entities with new matchId and null availability
+      Participant currParticipant =
+          participantDatastore.getParticipantFromUsername(match.getSecondParticipantUsername());
+      nullAvailabilityAdd(participantDatastore, currParticipant, matchId);
+      nullAvailabilityAdd(participantDatastore, newParticipant, matchId);
     } else {
```

Comment:
Now that user data is kept separate is this still needed?

Reply:
Yes, we need to store the matchId until the match is returned in SearchMatchServlet

Reply:
Got it, thanks